server/controllers/event.js

```diff
@@ -0,0 +1,25 @@
+const Event = require("../models/Event");
+
+// @route POST /event
+// @desc create an entry for the Event in the database
+// @access Private
+
+// Need the data in json foramt and date string should end with z
+// if not UTC time will be used as offset 
+const event = async (req, res) => {
+    const date = new Date(req.body.eventDate);
+    const eventRecord = new Event({
+        userURL : req.body.userURL,
+        eventDuration : req.body.eventDuration,
+        eventDate : date
+    });
+    try{
+        const savedPost = await eventRecord.save();
+        res.json(`Successfully created the event ${savedPost}`);
+    }catch(err){
+        res.json({message: err});
```

Comment:
Will this throw an error code or a status of 200?

Reply:
I forgot we are using express-async-handler! We can complete take the try and catch out of the code!
I see a scenarios where we need to pay attention too, 
1. If the body of the request have missing data (More appropriate)
- Due to express-async-handler we are getting status code 200, but the body contains the error details
- I can modify the content of ("./middlewear/error.js") to return the status code 422, if the error message contains "event validation"